examples/caa/caa.ipynb

```diff
@@ -0,0 +1,766 @@
+    "model, tokenizer = get_model_and_tokenizer(model_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up Datasets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For this demonstration, we'll download the sycophancy train and test split used in the CAA paper. CAA uses data formatted in the style of Anthropic's Model-Written Evals"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "--2024-01-28 11:38:42--  https://raw.githubusercontent.com/nrimsky/CAA/main/preprocessed_data/generate_dataset.json\n",
+      "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...\n",
+      "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443... connected.\n",
+      "HTTP request sent, awaiting response... 200 OK\n",
+      "Length: 804093 (785K) [text/plain]\n",
+      "Saving to: ‘generate_dataset.json.1’\n",
+      "\n",
+      "generate_dataset.js 100%[===================>] 785.25K  --.-KB/s    in 0.01s   \n",
+      "\n",
+      "2024-01-28 11:38:43 (79.2 MB/s) - ‘generate_dataset.json.1’ saved [804093/804093]\n",
+      "\n",
+      "--2024-01-28 11:38:43--  https://raw.githubusercontent.com/nrimsky/CAA/main/preprocessed_data/test_dataset.json\n",
+      "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.110.133, 185.199.108.133, 185.199.109.133, ...\n",
+      "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.110.133|:443... connected.\n",
+      "HTTP request sent, awaiting response... 200 OK\n",
+      "Length: 92183 (90K) [text/plain]\n",
+      "Saving to: ‘test_dataset.json.1’\n",
+      "\n",
+      "test_dataset.json.1 100%[===================>]  90.02K  --.-KB/s    in 0.003s  \n",
+      "\n",
+      "2024-01-28 11:38:43 (33.2 MB/s) - ‘test_dataset.json.1’ saved [92183/92183]\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Download the CAA dataset\n",
+    "!wget https://raw.githubusercontent.com/nrimsky/CAA/main/preprocessed_data/generate_dataset.json\n",
+    "!wget https://raw.githubusercontent.com/nrimsky/CAA/main/preprocessed_data/test_dataset.json"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import random\n",
+    "\n",
+    "# NOTE(dtch1997): I considered defining MWEData to be a dataclass instead\n",
+    "# which would make it clear what fields are expected and enable type-checking\n",
+    "# but I'm worried it will be confusing to the average researcher\n",
+    "# and I'd like to be consistent with Anthropic's MWE style\n",
+    "\n",
+    "MWEData = list[dict[str, str]]\n",
```

Comment:
Totally fair and I'm not really sure about this as well. I'm leaning towards keeping the type hints in, just because I've often gotten burned trying to read / use other people's code and being mega confused as to what functions were returning (especially if the variable name was not helpful either). I think it'd be nice if the community adopted this practice more widely :)